source/features/linkify-action.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'dom-chef';
+
+import * as pageDetect from 'github-url-detection';
+
+import {ArrowLeftIcon} from '@primer/octicons-react';
+
+import features from '../feature-manager.js';
+import api from '../github-helpers/api.js';
+import observe from '../helpers/selector-observer.js';
+
+async function getPullRequestNumber(user: string, repository: string, runDatabaseId: string): Promise<number | false> {
+	const {resource} = await api.v4(`
+		query getPullRequestNumber($resource: URI!) {
+			resource(url: $resource) {
+				... on WorkflowRun {
+					event
+					checkSuite {
+						matchingPullRequests(last: 1) {
+							edges {
+								node {
+									number
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	`, {
+		variables: {
+			resource: `/${user}/${repository}/actions/runs/${runDatabaseId}`,
+		},
+	});
+
+	if (resource.event !== 'pull_request') {
+		return false;
+	}
+
+	const {edges} = resource.checkSuite.matchingPullRequests;
+	if (edges.length === 0) {
+		return false;
+	}
+
+	return edges[0].node.number;
```

Comment:
You can use this format from now on, perhaps also changing your previous PR (which uses a similar .length check, if I remember correctly). That change can also be done in this PR

```suggestion
	return resource.checkSuite.matchingPullRequests.edges[0]?.node.number ?? false;
```